lib/engine/player.rb

```diff
@@ -41,7 +41,7 @@ def corporation
     end
 
     def ==(other)
-      other&.player? && (@name == other&.name)
+      other&.player? && !other.player.nil? && (@name == other.name)
```

Comment:
why was this needed?

Reply:
you shouldn't call nil? other.playre is enough

Reply:
As this discussion: https://18xxgames.slack.com/archives/C012K0CNY5C/p1708883331307309?thread_ts=1708871941.895459&cid=C012K0CNY5C
the issue is that in Javascript this code `other&.player?`  is translated into a check that player is not null, but in that case the Player is NilClass, so it's returning true also if the player is null
With this fix, the problem is solved also on JS

Reply:
can you do 

`!!other&.player? && name == other.name`

Reply:
done in d293b0f